lib/service.go

```diff
@@ -29,17 +28,23 @@ type BoostService struct {
 	getHeaderClient http.Client
 }
 
-func newBoostService(relayURLs []string, log *logrus.Entry) (*BoostService, error) {
+func newBoostService(relayURLs []string, log *logrus.Entry, getHeaderTimeout time.Duration) (*BoostService, error) {
 	if len(relayURLs) == 0 || relayURLs[0] == "" {
 		return nil, errors.New("no relayURLs")
 	}
 
+	// Set custom GetHeader timeout
+	_getHeaderTimeout := defaultGetHeaderTimeout
```

Comment:
Seems like you could get rid of this assignment by directly setting `getHeaderTimeout` if it is `0`?
```go
if getHeaderTimeout == 0 {
    getHeaderTimeout = defaultGetHeaderTimeout
}
```

Reply:
nice, done 🙏